zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -558,6 +559,33 @@ mod tests {
         }
 
         let expected_views = vec![
+            // v_sapling_shard_unscanned_ranges
+            format!(
+                "CREATE VIEW v_sapling_shard_unscanned_ranges AS
+                SELECT
+                    shard.shard_index,
+                    shard.shard_index << 16 AS start_position,
+                    (shard.shard_index + 1) << 16 AS end_position_exclusive,
+                    IFNULL(prev_shard.subtree_end_height, {}) AS subtree_start_height,
+                    shard.subtree_end_height AS subtree_end_height,
+                    shard.contains_marked,
+                    scan_queue.block_range_start,
+                    scan_queue.block_range_end,
+                    scan_queue.priority
+                FROM sapling_tree_shards shard
+                LEFT OUTER JOIN sapling_tree_shards prev_shard
+                    ON shard.shard_index = prev_shard.shard_index + 1
+                INNER JOIN scan_queue ON 
+                    (scan_queue.block_range_start BETWEEN subtree_start_height AND shard.subtree_end_height) OR
```

Comment:
The naming problem is hard here. It could be that `subtree_last_position_height` is a better name, but the query is inclusive on both ends because both the first and last position of notes within a subtree may be in the middle of blocks. It's a wart that this view is needed at all, really; had we waited longer to release the updated `lightwalletd` we might have decided to include the start height information in the shard metadata directly, and so wouldn't have to compute it and could have just done a normal join against the `sapling_tree_shards` table.